pages/impersonate-user.js

```diff
@@ -0,0 +1,168 @@
+// node_modules
+import { useRouter } from "next/router";
+import PropTypes from "prop-types";
+import { useContext, useState } from "react";
+// components
+import useDebounceTimer from "../components/debounce-timer";
+import { Button } from "../components/form-elements";
+import NoCollectionData from "../components/no-collection-data";
+import PagePreamble from "../components/page-preamble";
+import SessionContext from "../components/session-context";
+import { TextField } from "../components/form-elements";
+// lib
+import errorObjectToProps from "../lib/errors";
+import FetchRequest, { HTTP_STATUS_CODE } from "../lib/fetch-request";
+import { encodeUriElement } from "../lib/query-encoding";
+
+/**
+ * Send a request to the data provider to impersonate the user with the given @id.
+ * @param {string} userId @id of the user to impersonate
+ * @param {object} session Session object from SessionContext
+ * @returns
+ */
+async function requestImpersonateUser(userId, session) {
+  const request = new FetchRequest({ session });
+  return request.postObject("/impersonate-user", { user: userId });
```

Comment:
Why do we need to post here: /impersonate-user? Can we post directly at home page?

Reply:
That’s the igvfd endpoint for triggering impersonation. You post an object with the user’s `@id` to /impersonate-user. Posting anything to "/" wouldn’t do anything.

Reply:
Oh, got it. Thanks!